Samples/StyleTransfer/AppViewModel.cs

```diff
@@ -0,0 +1,546 @@
+        private SolidColorBrush _failColor = new SolidColorBrush(Windows.UI.Colors.Red);
+
+        private AppModel _appModel;
+        public AppModel CurrentApp
+        {
+            get { return _appModel; }
+        }
+        public float RenderFPS
+        {
+            get { return (float)Math.Round(_renderFPS, 2); }
+            set { _renderFPS = value; OnPropertyChanged(); }
+        }
+
+        private float _captureFPS;
+        public float CaptureFPS
+        {
+            get
+            { return _captureFPS; }
+            set
+            { _captureFPS = value; OnPropertyChanged(); }
+        }
+        private bool _useGpu;
+        public bool UseGpu
+        {
+            get { return _useGpu; }
+            set
+            {
+                _useGpu = value;
+                if (_appModel.InputMedia == "LiveStream")
+                {
+                    SetMediaSourceCommand.Execute("LiveStream");
+                }
+                OnPropertyChanged();
+            }
+        }
+        private SoftwareBitmapSource _inputSoftwareBitmapSource;
+        public SoftwareBitmapSource InputSoftwareBitmapSource
+        {
+            get { return _inputSoftwareBitmapSource; }
+            set { _inputSoftwareBitmapSource = value; OnPropertyChanged(); }
+        }
+        private SoftwareBitmapSource _outputSoftwareBitmapSource;
+        public SoftwareBitmapSource OutputSoftwareBitmapSource
+        {
+            get { return _outputSoftwareBitmapSource; }
+            set { _outputSoftwareBitmapSource = value; OnPropertyChanged(); }
+        }
+        private bool _saveEnabled;
+        public bool SaveEnabled
+        {
+            get { return _saveEnabled; }
+            set { _saveEnabled = value; OnPropertyChanged(); }
+        }
+
+        public SolidColorBrush StatusBarColor
+        {
+            get { return _succeed ? _successColor : _failColor; }
+        }
+
+        private string _notifyMessage;
+        public string StatusMessage
+        {
+            get { return _notifyMessage; }
+            set { _notifyMessage = value; OnPropertyChanged(); OnPropertyChanged("StatusBarColor"); }
+        }
+
+        // Command defintions
+        public ICommand SaveCommand { get; set; }
+        public ICommand SetMediaSourceCommand { get; set; }
+        public ICommand ChangeLiveStreamCommand { get; set; }
+        public ICommand SetModelSourceCommand { get; set; }
+
+        public async void SaveOutput()
+        {
+            Debug.WriteLine("UIButtonSaveImage_Click");
+            if (_appModel.InputMedia != "LiveStream") await ImageHelper.SaveVideoFrameToFilePickedAsync(_appModel.OutputFrame);
+        }
+
+        public async Task SetMediaSource(string src)
+        {
+            _appModel.InputMedia = src;
+            await CleanupCameraAsync();
+            CleanupInputImage();
+
+            NotifyUser(true);

```

Comment:
>NotifyUser(true); [](start = 12, length = 17)

Why notify success here?

Reply:
In case a previous display had an error, resets the status bar (eg. the last time you uploaded a picture there was an issue, but this time around you successfully upload)

Reply:
shouldnt this happen before LoadModelAsync?

---
In reply to: [468736416](https://github.com/microsoft/Windows-Machine-Learning/pull/335#discussion_r468736416) [](ancestors = 468736416)